ably-ios/ARTRealtimePresence.m

```diff
@@ -48,33 +72,45 @@ - (void)enterClient:(NSString *)clientId data:(id)data cb:(ARTStatusCallback)cb
     msg.clientId = clientId;
     msg.data = data;
 
-    msg.connectionId = [self channel].realtime.connectionId;
+    msg.connectionId = [self channel].realtime.connection.id;
     [[self channel] publishPresence:msg cb:cb];
 }
 
-- (void)update:(id)data cb:(ARTStatusCallback)cb {
+- (void)update:(id)data {
+    [self update:data cb:nil];
+}
+
+- (void)update:(id)data cb:(void (^)(ARTErrorInfo * _Nullable))cb {
     [self updateClient:[self channel].clientId data:data cb:cb];
 }
 
-- (void)updateClient:(NSString *)clientId data:(id)data cb:(ARTStatusCallback)cb {
+- (void)updateClient:(NSString *)clientId data:(id)data {
+    [self updateClient:clientId data:data cb:nil];
+}
+
+- (void)updateClient:(NSString *)clientId data:(id)data cb:(void (^)(ARTErrorInfo * _Nullable))cb {
     ARTPresenceMessage *msg = [[ARTPresenceMessage alloc] init];
     msg.action = ARTPresenceUpdate;
     msg.clientId = clientId;
-    if(!msg.clientId) {
```

Comment:
`updateClient` takes a non-nullable clientId as argument. The compiler will catch this. In any case, we can't be adding null checks everywhere.